### PR TITLE
refactor packaging_build_rpm

### PR DESCRIPTION
**Pull request by komidore64**
will happily squash upon acceptance

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    stages {
+        stage("Setup Environment") {
+            steps {
+                git(url: 'https://github.com/theforeman/foreman-packaging/', branch: env.branch)
+                setup_obal()
+            }
+        }
+
+        stage("Apply Pull Request Changes") {
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: Up until 1.17 we used the packaging_build_rpm job for foreman-packaging PRs as well but we no longer do that. I think that nothing calls this task with a PR anymore and you can remove this.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        build_package_tito_releaser_args += " --arg jenkins_job_id=${env.nightly_jenkins_job_id}"
+                    }
+
+                    obal(
+                        action: 'release',
+                        extraVars: [
+                            'build_package_tito_releaser_args': [build_package_tito_releaser_args],
+                            'build_package_scratch': build_package_scratch,
+                            'build_package_test': build_package_test,
+                            'releaser': env.releaser
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: Could we ignore this and rely on obal to do this? If we use `if (nightly) { obal nightly } else { obal release }` we should be there. Then you can drop the `_scratch` and `_test` args I think.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy [hunk 3]
```diff
@@ -0,0 +1,77 @@
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        disableConcurrentBuilds()
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: Previously we had concurrent builds. Any reason to disable this now?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                    expression { env.pr_git_url != null && env.scratch == 'true' }
+                    expression { env.pr_git_url == null }
+                }
+            }
+            steps {
+                script {
+                    sh("mkdir -p ${env.WORKSPACE}/rel-eng/build")
+
+                    def build_package_tito_releaser_args = "-o ${env.WORKSPACE}/rel-eng/build/"
+                    def build_package_scratch = false
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: Nothing calls this job for scratch builds. Again a leftover from the PR testing.

**Comment by ehelms** [comment]
I thought we wanted to move away from generic jobs, to specific jobs but using library functions to re-use code.

**Comment by ekohl** [comment]
> I thought we wanted to move away from generic jobs, to specific jobs but using library functions to re-use code.

I thought so too, but figured this was an intermediate step.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy (hunk 3 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: Whoops, copy/paste I suppose.

**Comment by komidore64** [comment]
Yes this is an intermediate step. It's much easier on my brain to go through a transition period and keep things continually working as much as possible. I don't want to have a gigantic PR rewriting all the nightly stuff at once.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml [hunk 5]
```diff
@@ -1,15 +1,7 @@
-# Builds a single Foreman (sub)project or dependency from the foreman-packaging
-# repository, using Koji.
-#
-# See https://projects.theforeman.org/projects/foreman/wiki/RPM_Packaging#Project-sources-version-3
-# or the help text on the build page for usage.
 - job:
     name: packaging_build_rpm
-    concurrent: true
```
Comment (ehelms) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml]: You probably want to keep this given our recent learning.

Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml]: urite

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy (hunk 2 above)
Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: to support calling `obal nightly` versus calling `obal release`, i need to straight out theforeman/obal#125 first.

File: puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        stage("Build RPM") {
+            steps {
+                script {
+                    sh("mkdir -p ${env.WORKSPACE}/rel-eng/build")
+
+                    def build_package_tito_releaser_args = "-o ${env.WORKSPACE}/rel-eng/build/"
+
+                    if (env.tag) {
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: I wonder if we use this anywhere. I don't see it in jobs and we don't use `tito tag` anymore so I think you can remove it.

Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/packaging_build_rpm.groovy]: perfect. more things to delete!

**Comment by komidore64** [comment]
updated!

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml [hunk 7]
```diff
@@ -1,55 +1,13 @@
-# Builds a single Foreman (sub)project or dependency from the foreman-packaging
-# repository, using Koji.
-#
-# See https://projects.theforeman.org/projects/foreman/wiki/RPM_Packaging#Project-sources-version-3
-# or the help text on the build page for usage.
 - job:
     name: packaging_build_rpm
+    project-type: pipeline
+    sandbox: false
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml]: Is there a reason we can't sandbox?

Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml]: I remember in the past when I was converting to pipeline jobs `sandbox` wasn't what I thought it meant and it ended up making things not work.

Comment (komidore64) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/packaging_build_rpm.yaml]: well [this](https://docs.openstack.org/infra/jenkins-job-builder/project_pipeline.html?highlight=sandbox) isn't helpful:

```
sandbox (bool): If the script should run in a sandbox (default false)
```

**Comment by ehelms** [comment]
@ekohl Now that RC1 is out, are we ready to put this change into operation?

**Comment by ekohl** [comment]
Yep. Do note we're planning on releasing RC2 on Wednesday/Thursday so I'll be coming for you when things break then ;)

**Pull request merged**